agent/api/task/task.go

```diff
@@ -202,53 +206,59 @@ func (task *Task) PostUnmarshalTask(cfg *config.Config,
 			return apierrors.NewResourceInitError(task.Arn, err)
 		}
 	}
-	task.initializeEmptyVolumes()
+	err := task.initializeDockerLocalVolumes(dockerClient, ctx)
+	if err != nil {
+		return apierrors.NewResourceInitError(task.Arn, err)
+	}
 	task.initializeCredentialsEndpoint(credentialsManager)
 	task.addNetworkResourceProvisioningDependency(cfg)
 	return nil
 }
 
-func (task *Task) initializeEmptyVolumes() {
-	requiredEmptyVolumes := []string{}
+func (task *Task) initializeDockerLocalVolumes(dockerClient dockerapi.DockerClient, ctx context.Context) error {
+	requiredLocalVolumes := []string{}
```

Comment:
nit: `var requiredLocalVolumes []string` is the idiomatic way of doing this i think